pandas/_libs/tslibs/timezones.pyi

```diff
@@ -0,0 +1,32 @@
+from datetime import (
+    datetime,
+    tzinfo,
+)
+from typing import (
+    Callable,
+    Optional,
+    Union,
+)
+
+import numpy as np
+
+# imported from dateutil.tz
+dateutil_gettz: Callable[[str], tzinfo]
+
+
+def tz_standardize(tz: tzinfo) -> tzinfo: ...
+
+def tz_compare(start: Optional[tzinfo], end: Optional[tzinfo]) -> bool: ...
+
+def infer_tzinfo(
+    start: Optional[datetime], end: Optional[datetime],
+) -> Optional[tzinfo]: ...
+
+# ndarrays returned are both int64_t
+def get_dst_info(tz: tzinfo) -> tuple[np.ndarray, np.ndarray, str]: ...
+
+def maybe_get_tz(tz: Optional[Union[str, int, np.int64, tzinfo]]) -> Optional[tzinfo]: ...
+
+def get_timezone(tz: tzinfo) -> Union[tzinfo, str]: ...
```

Comment:
again from https://github.com/python/typeshed/blob/master/CONTRIBUTING.md#stub-file-coding-style

> avoid Union return types: https://github.com/python/mypy/issues/1693

although not sure this is possible without a refactor.

Reply:
yah id like to change this behavior too, but should be separate from typing